File: src/main/java/com/bugsnag/android/Notifier.java
package com.bugsnag.android;

import android.support.annotation.NonNull;

import java.io.IOException;

/**
 * Information about this library, including name and version.
 */
class Notifier implements JsonStream.Streamable {
    static final String NOTIFIER_NAME = "Android Bugsnag Notifier";
    static final String NOTIFIER_VERSION = "3.4.0";
    static final String NOTIFIER_URL = "https://bugsnag.com";

    private static final Notifier instance = new Notifier();
    public static Notifier getInstance() {
        return instance;
    }

    public void toStream(@NonNull JsonStream writer) throws IOException {
        writer.beginObject();
            writer.name("name").value(NOTIFIER_NAME);
            writer.name("version").value(NOTIFIER_VERSION);
            writer.name("url").value(NOTIFIER_URL);
        writer.endObject();
    }
}

File: bugsnag-android-core/src/main/java/com/bugsnag/android/Notifier.kt
package com.bugsnag.android

import com.bugsnag.android.internal.journal.JournalKeys
import com.bugsnag.android.internal.journal.Journalable
import java.io.IOException

/**
 * Information about this library, including name and version.
 */
class Notifier @JvmOverloads constructor(
    var name: String = "Android Bugsnag Notifier",
    var version: String = "5.15.0",
    var url: String = "https://bugsnag.com"
) : JsonStream.Streamable, Journalable {

    var dependencies = listOf<Notifier>()

    @Throws(IOException::class)
    override fun toStream(writer: JsonStream) = writer.value(toJournalSection())

    override fun toJournalSection(): Map<String, Any?> {
        val data = mutableMapOf(
            JournalKeys.keyName to name,
            JournalKeys.keyVersion to version,
            JournalKeys.keyUrl to url
        )
        return when {
            dependencies.isNotEmpty() -> data.plus(
                Pair("dependencies", dependencies.map(Notifier::toJournalSection))
            )
            else -> data
        }
    }
}
